Ginger/GingerCoreNET/Logger/WebReportGenerator.cs

```diff
@@ -129,61 +130,48 @@ public void DeleteFoldersData(string clientAppFolderPath)
-                    if (totalActivities != 0)
-                        liteDbBusinessFlow.ExecutionRate = string.Format("{0:F1}", (totalExecutedActivities * 100 / totalActivities).ToString());
-                    if (totalExecutedActivities != 0)
-                        liteDbBusinessFlow.PassRate = string.Format("{0:F1}", (totalPassedActivities * 100 / totalExecutedActivities).ToString());
+                    if (liteDbBusinessFlow.ChildExecutableItemsCount[_HTMLReportConfig.ExecutionCalculationLevel] != 0)
+                        liteDbBusinessFlow.ExecutionRate = string.Format("{0:F1}", (liteDbBusinessFlow.ChildExecutedItemsCount[_HTMLReportConfig.ExecutionCalculationLevel] * 100 / liteDbBusinessFlow.ChildExecutableItemsCount[_HTMLReportConfig.ExecutionCalculationLevel]).ToString());
+                    if (liteDbBusinessFlow.ChildExecutedItemsCount[_HTMLReportConfig.ExecutionCalculationLevel] != 0)
+                        liteDbBusinessFlow.PassRate = string.Format("{0:F1}", (liteDbBusinessFlow.ChildPassedItemsCount[_HTMLReportConfig.ExecutionCalculationLevel] * 100 / liteDbBusinessFlow.ChildExecutedItemsCount[_HTMLReportConfig.ExecutionCalculationLevel]).ToString());
                     if (liteDbBusinessFlow.Elapsed.HasValue)
                         liteDbBusinessFlow.Elapsed = Math.Round(liteDbBusinessFlow.Elapsed.Value, 2);
                     foreach (LiteDbActivity liteDbActivity in liteDbBusinessFlow.ActivitiesColl)
                     {
-                        int totalActions = liteDbActivity.ActionsColl.Count;
-                        int totalPassedActions = liteDbActivity.ActionsColl.Where(ac => ac.RunStatus == eRunStatus.Passed.ToString()).Count();
-                        int totalExecutedActions = totalActions - liteDbActivity.ActionsColl.Where(ac => ac.RunStatus == eRunStatus.Pending.ToString() || ac.RunStatus == eRunStatus.Skipped.ToString() || ac.RunStatus == eRunStatus.Blocked.ToString()).Count();
-                        if (totalActions != 0)
-                            liteDbActivity.ExecutionRate = string.Format("{0:F1}", (totalExecutedActions * 100 / totalActions).ToString());
-                        if (totalExecutedActions != 0)
-                            liteDbActivity.PassRate = string.Format("{0:F1}", (totalPassedActions * 100 / totalExecutedActions).ToString());
+                        if (liteDbActivity.ChildExecutableItemsCount[_HTMLReportConfig.ExecutionCalculationLevel] != 0)
+                            liteDbActivity.ExecutionRate = string.Format("{0:F1}", (liteDbActivity.ChildExecutedItemsCount[HTMLReportConfiguration.ReportsExecutionCalculationLevel.Action.ToString()] * 100 / liteDbActivity.ChildExecutableItemsCount[HTMLReportConfiguration.ReportsExecutionCalculationLevel.Action.ToString()]).ToString());
```

Comment:
for Activity you have only "Actions" values and you always need to calculate based on it, even if user select to calculate based on "Activities", it can't be zero/empty if user select "Activitie" based calculations...